unique_toolkit/tests/language_model/test_language_models_infos.py

```diff
@@ -63,3 +64,25 @@ def test_get_custom_language_model(self):
     def test_get_language_model_raises_error_for_invalid_model(self):
         with pytest.raises(ValueError):
             LanguageModel("")
+
+    # New tests for LanguageModelTokenLimits
+    def test_language_model_token_limits_with_input_output(self):
+        limits = LanguageModelTokenLimits(
+            token_limit_input=2000, token_limit_output=2000
+        )
+        assert limits.token_limit == 4000
+        assert abs(limits.fraction_input - 0.5) < 1e-10
+
+    def test_language_model_token_limits_with_total(self):
+        limits = LanguageModelTokenLimits(token_limit=10000)
+        assert limits.token_limit_input == 4000.0
+        assert limits.token_limit_output == 6000.0
+
+    def test_language_model_token_limits_with_total_and_fraction(self):
+        limits = LanguageModelTokenLimits(token_limit=10000, fraction_input=0.2)
+        assert limits.token_limit_input == 2000.0
+        assert limits.token_limit_output == 8000.0
+
+    def test_language_model_token_limits_raises_error(self):
+        with pytest.raises(ValueError):
+            LanguageModelTokenLimits()
```

Comment:
should add more tests where only `token_limit_input`  or `token_limit_output` is provided.

Reply:
Thanks, it's done